File: python/leetcode/longest-palindrome.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        seen = {}
        for c in s:
            seen[c] = seen[c] + 1 if c in seen else 1
        evens = 0
        odds = 0
        for k in seen:
            if seen[k] % 2 == 0:
                evens += seen[k]
            else:
                evens += seen[k] - 1
                odds = 1
        return evens + odds

File: java/leetcode/longest-palindrome.java
class Solution {
    static int longestPalindrome(String s) {
    	HashMap<Character, Integer> seen = new HashMap<Character, Integer>();
    	for (int i = 0; i < s.length(); i++) {
    		seen.put(s.charAt(i), seen.getOrDefault(s.charAt(i), 0) + 1);
    	}
    	int evens = 0;
    	int odds = 0;
    	Iterator keys = seen.entrySet().iterator();
    	while (keys.hasNext()) {
    		Map.Entry<String, Integer> pair = (Map.Entry)keys.next();
    		if (pair.getValue() % 2 == 0) {
    			evens += pair.getValue();
    		} else {
    			evens += pair.getValue() - 1;
    			odds = 1;
    		}
    	}
    	return evens + odds;
    }
}